### PR TITLE
Using Bootstrap: The newsweek magazine Clone

### PR DESCRIPTION
**Pull request by keddo**
**Building Responsive Website Using Frameworks (Bootstrap)**

This project requests  to build a replica of the news site newsweek.com using the Bootstrap framework.
  

-   With some custom CSS
-   Built with bootstrap
-   Added some Javascript[optional]
-   Built with SASS which is also optional
-   Responsive from small to large screen sizes

I am kindly requesting for review.
Thank you.

### CODE REVIEW

**Review by trekab — COMMENTED** [review]
# Status: Invalid :stop_sign: 
Hi @keddo, it is unfortunate that your code review request is invalid because you didn't provide a meaningful name for your Pull Request and you have no short summary of changes introduced in the Pull Request. ([Professional Github repository](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243642/)). Please make the necessary changes and submit another code review request.

_Feel free to reach me on slack for any inquiries about this_ :wink: 
__Treasure Kabareebe__

**Review by sumancrest0001 — COMMENTED** [review]
**STATUS: CHANGES REQUIRED
Hello @keddo,**
You have done a fantastic job to make it very clone to the original one. Your project looks almost perfect to me and most of the minimal requirements are ticked. But there are a couple of minor issues that need to be addressed to get :+1:  from our side. I am pretty confident that it won't take long to fix them. Please make the following changes.

[REQUIRED] There is no Newsweek logo at the footer section.
Currently :point_down: 
Screenshot from 2020-05-30 13-58-30

File: index.html [hunk 1]
```diff
@@ -0,0 +1,2378 @@
...
+                        </span>
+                    </div>
+                    <div class="date-info text-white">
+                        <time datetime="2020-05-27">Wed, May 20, 2020</time>
+                    </div>
+                </div>
+                <div class="header-logo">
+                    <a class="brand" href="https://www.newsweek.com">
+                        <svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg"
+                            xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px" viewBox="0 0 395.6 60"
```
Comment (sumancrest0001) [inline index.html:47]: [REQUIRED] We have to fix all the error shown by the HTML validator. I am using [this](https://validator.w3.org) validator.

File: assets/css/main.css [hunk 2]
```diff
@@ -0,0 +1,996 @@
...
+h4,
+h5,
+h6 {
+  font-family: "Roboto Condensed", sans-serif !important;
+}
+
+.nav-header .header-top {
+  background-color: #f72210 !important;
+  height: 95px;
+  padding: 20px 10px 15px;
```
Comment (sumancrest0001) [inline assets/css/main.css]: [REQUIRED] Your CSS file looks good but you are using the `!important` 29 times. We should avoid using !important as much as we could. It is not a good practice to over use it. Let's remove it from the styling.

**Review by sumancrest0001 — COMMENTED** [review]
[Optional] On the mobile screen, the following image is not taking full width. I encourage you to fix this.
Screenshot from 2020-05-30 14-01-20

**Review by sumancrest0001 — CHANGES_REQUESTED** [review]
**Kindly make the requested changes and request for another code review.

If you have any query regarding the review, feel free to message me on Slack Suman Shrestha
Happy coding !!!**

File: assets/css/main.css (hunk 2 above)
Comment (keddo) [reply assets/css/main.css]: I am using !important because of bootstrap I can't overide bootstrap class without using it

File: index.html (hunk 1 above)
Comment (keddo) [reply index.html:47]: - HTML validation shows up as follows:  
error-check

**Review by sumancrest0001 — COMMENTED** [review]
**STATUS: Changes required
Hello @keddo,**
Your project layout looks very similar to the original one and it is responsive too. But we still need to get rid of `! important` as it is not a good practice. I agree with you that without `!important` we can not override the default stylings. To solve this, instead of overriding bootstrap classes you can define custom classes and add them in the HTML.

**Review by sumancrest0001 — COMMENTED** [review]
Kindly make the requested change and submit for another review.
[Note] I encourage to remove most of them if not all.
I have also sent a message on slack to you. We can discuss it over there if you want.

**Happy coding!!!**

**Comment by keddo** [comment]
Changes Made:
  - I removed more than half of !important keyword from css
  - But I leave the rest because the it depends on media queries and have a lot of markup elements 

Thank you.

**Review by rammazzoti2000 — CHANGES_REQUESTED** [review]
### Hello @keddo :wave: I am Alex your code reviewer for this project
Great work on the project, good work on impementing the changes required by previous TSEs.
There are still some changes left from the last tse (@sumancrest0001) that need to be fix before approving this project, see the comments bellow :point_down: 
### :stop_sign: Required Changes :construction: :construction_worker: 
- [x] Your HTML code didn't pass validation, consider using [this validator](https://validator.w3.org/#validate_by_input) by direct input
image
- [x] Your `main.css` file combine with all `.scss` files from sass folder have in total 43 `!important` properties. Let's consider only the `main.css` file you still have 14 `!important` properties which are still a lot. It could be allowed  a few but not 14 and for sure not 43. These `!important` properties are mainly used because you didn't make full use of bootstrap.
:exclamation: Remeber, this is a bootstrap project. Your main focus as I can notice from your project was centered around sass and css styling. This projects only needs an html file with bootstrap classes and just some styling in css, but not more.
##
I highly recommend you to learn more about the CSS specificity which will give you help you to learn which style declaration will be the winner and why. It will help you to decide what you should each time to override a Bootstrap style :point_right: [CSS Specificity - W3Schools](https://www.w3schools.com/css/css_specificity.asp) :exclamation: 
##
Consider removing most of the `!important` properties if not all of them before submitting for a new review.
##
### <em>Kindly fix the REQUIRED issues and send another code review request.

Cheers and Happy coding! :clap: 

Code review by Alexandru Bangau

(Feel free to contact me on slack by the handle of **_alexandru_** if you have any additional question)</em>
### :stop_sign: Status: Required Changes

File: assets/css/main.css [hunk 3]
```diff
@@ -0,0 +1,996 @@
...
+
+main .article-category {
+  position: absolute;
+  left: 0;
+  bottom: 0;
+  z-index: 3;
+}
+
+.article-title .text-dark {
+  color: #000 !important;
```
Comment (rammazzoti2000) [inline assets/css/main.css:257]: this color can be used as `.text-dark` bootstrap class

File: assets/css/main.css [hunk 4]
```diff
@@ -0,0 +1,996 @@
...
+    order: -1;
+  }
+
+  .culture-item {
+    padding-left: 0;
+    padding-right: 0;
+  }
+
+  .culture-stories {
+    padding-top: 0 !important;
```
Comment (rammazzoti2000) [inline assets/css/main.css]: `pt-md-0` --> @media (min-width: 768px) == md

File: assets/css/main.css [hunk 5]
```diff
@@ -0,0 +1,996 @@
...
+  }
+
+  .culture-items {
+    margin-left: 0;
+    margin-right: 0;
+  }
+
+  .section-picks article.py-3 {
+    padding-bottom: 0 !important;
```
Comment (rammazzoti2000) [inline assets/css/main.css]: `pb-md-0` --> bootstrap class

File: assets/css/main.css [hunk 6]
```diff
@@ -0,0 +1,996 @@
...
+    padding-top: 0 !important;
```
Comment (rammazzoti2000) [inline assets/css/main.css]: `pt-md-0` --> bootstrap class

File: assets/css/main.css [hunk 7]
```diff
@@ -0,0 +1,996 @@
...
+    height: 100%;
+  }
+
+  .subscribe-block .f1 {
+    width: 80%;
+    text-align: left;
+  }
+
+  .subscribe-block .col-l .image.last {
+    padding-right: 0 !important;
```
Comment (rammazzoti2000) [inline assets/css/main.css]: `pr-md-0` --> bootstrap class

File: assets/css/main.css [hunk 8]
```diff
@@ -0,0 +1,996 @@
...
+}
+
+@media (min-width: 993px) {
+  .container-fluid {
+    max-width: 1310px;
+    padding: 0 30px;
+  }
+
+  .nav-header .container-fluid {
+    padding-left: 0 !important;
```
Comment (rammazzoti2000) [inline assets/css/main.css]: `pl-lg-0` --> @media (min-width: 992px) == lg

File: assets/css/main.css [hunk 9]
```diff
@@ -0,0 +1,996 @@
...
+    padding-right: 0 !important;
```
Comment (rammazzoti2000) [inline assets/css/main.css]: `pr-lg-0` --> bootstrap class

File: assets/css/main.css [hunk 10]
```diff
@@ -0,0 +1,996 @@
...
+  }
+
+  .d-pt {
+    padding-top: 1.5rem !important;
+    padding-bottom: 1.5rem !important;
```
Comment (rammazzoti2000) [inline assets/css/main.css]: `pt-lg-2 pb-lg-2` --> bootstrap classes

File: assets/css/main.css (hunk 5 above)
Comment (keddo) [reply assets/css/main.css]: done!

File: assets/css/main.css (hunk 4 above)
Comment (keddo) [reply assets/css/main.css]: Done!

File: assets/css/main.css (hunk 3 above)
Comment (keddo) [reply assets/css/main.css:257]: .text-dark don't make the text black, It makes it light black instead.

File: assets/css/main.css (hunk 9 above)
Comment (keddo) [reply assets/css/main.css]: Done!

File: assets/css/main.css (hunk 6 above)
Comment (keddo) [reply assets/css/main.css]: done!

File: assets/css/main.css (hunk 7 above)
Comment (keddo) [reply assets/css/main.css]: Done

File: assets/css/main.css (hunk 8 above)
Comment (keddo) [reply assets/css/main.css]: Done!

File: assets/css/main.css (hunk 10 above)
Comment (keddo) [reply assets/css/main.css]: Done!

Comment (keddo) [reply assets/css/main.css]: **Changes made to requests above:**
   - Fix HTML validation error
   - Use bootstrap classes to reduce the number of !important keyword used.

**Comment by keddo** [comment]
**Changes made to requests above:**

- Fix HTML validation error
- Use bootstrap classes to reduce the number of !important keyword used.

**Review by rammazzoti2000 — APPROVED** [review]
### :heavy_check_mark: Project Approved :smile: :tada: 
Congratulations @keddo the project looks good and is ready for approval, also congrats for implementing the changes required before. Please feel free to merge whenever you feel ready.
##
<em>Cheers and Happy coding! :clap: 

Code review by Alexandru Bangau

(Feel free to contact me on slack by the handle of alexandru if you have any additional question)</em>
### :heavy_check_mark: Status: Project Approved

File: assets/css/main.css (hunk 3 above)
Comment (rammazzoti2000) [reply assets/css/main.css:257]: Alright no problem with that :smile:

**Pull request merged**